sdks/python/apache_beam/io/gcp/datastore/v1new/datastoreio.py

```diff
@@ -395,17 +416,34 @@ def write_mutations(self, throttler, rpc_stats_callback, throttle_delay=1):
         for element in self._batch_elements:
           self.add_to_batch(element)
 
+      # Create request count metric
+      resource = resource_identifiers.DatastoreNamespace(self._project, None)
+      labels = {
+          monitoring_infos.SERVICE_LABEL: 'Datastore',
+          monitoring_infos.METHOD_LABEL: 'BatchDatastoreWrite',
+          monitoring_infos.RESOURCE_LABEL: resource,
+          monitoring_infos.DATASTORE_NAMESPACE_LABEL: None,
+          monitoring_infos.DATASTORE_PROJECT_ID_LABEL: self._project,
+          monitoring_infos.STATUS_LABEL: 'ok'
+      }
+
+      service_call_metric = ServiceCallMetric(
+          request_count_urn=monitoring_infos.API_REQUEST_COUNT_URN,
+          base_labels=labels)
+
       try:
         start_time = time.time()
         self._batch.commit()
         end_time = time.time()
+        service_call_metric.call('ok')
 
         rpc_stats_callback(successes=1)
         throttler.successful_request(start_time * 1000)
         commit_time_ms = int((end_time - start_time) * 1000)
         return commit_time_ms
-      except Exception:
+      except Exception as e:
         self._batch = None
+        service_call_metric.call(e.code)
```

Comment:
This won't work. The base Exception class does not have a .code attribute.
Please catch the exception properly, using the specific exception type or otherwise determine how to get the grpc/http error code out of this client API and pass it in here.

Please ask on the beam dev list for a way to run an integration test and make sure this works through testing.
Log that you can pull out the exception
Make sure it doesn't raise another exception(i.e. attribute error) when this exception branch is ran

Reply:
Change done! I was able to run the integration test to catch and log the exception 

<img width="517" alt="Screen Shot 2021-08-16 at 11 25 20" src="https://user-images.githubusercontent.com/74670721/129607652-79acb0f4-33d1-4a41-816f-51e43cc0e458.png">